website/content/labs/lab6.md

```diff
@@ -0,0 +1,176 @@
+
+Note: Before committing your `.gitlab-ci.yml` file, you can check that the format of the file is correct by going to https://mit.cs.uchicago.edu/ci/lint. This will allow you to copy-paste your `.gitlab-ci.yml` file and check it for syntax errors.
+
+
+
+# Task 2: Adding tests
+
+As we mentioned previously, CI typically involves running tests every time we push changes to the repository. This means that, besides building our code, we also want tests to run automatically. For the simple hello program, we’ve included a very simple test: we simply run the program (using the `tests` target in the Makefile). While this doesn’t guarantee that the program prints the correct output, it will at least verify that the program runs without errors.
+
+So, let’s add a job to the `.gitlab-ci.yml` file to run the tests:
+
+    hello:test:
+        stage: test
+        script:
+          - make -C lab6/hello clean tests
+        only: 
+          - master
+
+Notice how the stage is `test`. This means that GitLab will not run this job unless all the jobs in the `build` stage succeed.
+
+Now, do the following:
+
+* [5 points] Commit the updated `.gitlab-ci.yml` file, and copy-paste the URL of the resulting “passed” pipeline to the `tasks.txt` file. Notice how GitLab now shows a `hello:test` job that depends on the build jobs. If you click on the `hello:test` job and look at its terminal output, you will see the output of running `hello`.
+* [10 points] Edit `hello.c` to force it to crash (e.g., dereference a NULL pointer). This will still allow `hello.c` to compile, but will make the tests fail. Commit and push this change, and copy-paste the URL of the resulting “failed” pipeline to the `tasks.txt` file. Notice how the build jobs ran successfully, but the pipeline failed because one of the jobs (the `hello:test` job) failed. Before moving on to the next test, undo this change and make sure it results in a “passed” pipeline.
+* [25 points] Add a `libgeometry:test` job to the `.gitlab-ci.yml` file to build and run the tests. This job should be in the `test` stage, even though it involves building code (in this case, building the tests). Furthermore, your `script` option will now have more than one entry: one for building the tests, and one or more to run them. Once you’re done, copy-paste the URL of the “passed” pipeline to the tasks.txt file.
+
+Careful: *Figuring out the commands for running the libgeometry tests is non-trivial!* Here are two hints: 
+* You should first figure out the exact commands you need to run from the root of your repository to successfully run the tests. **DO NOT** figure out the commands by pushing a new commit (and triggering a new pipeline) for every command you try to run. If you figure out the commands from the root of your repository, you should be able to then simply copy/paste those commands into your GitLab CI file.
```

Comment:
Should we explicitly tell them about `make -C`? Or at least point them to the `make` manpage?